rust/kernel/file.rs

```diff
@@ -12,7 +12,7 @@ use core::{mem::ManuallyDrop, ops::Deref};
 ///
 /// # Invariants
 ///
-/// The pointer [`File::ptr`] is non-null and valid. Its reference count is also non-zero.
+/// The pointer `File::ptr` is non-null and valid. Its reference count is also non-zero.
```

Comment:
I think you can use `File.ptr` as link.

Reply:
That works, but the link appears still as "[`File.ptr`]" i.e. it is still broken because it points to a private field. :-(

Reply:
The other day we were discussing whether the invariants should be public or hide them, or whether we wanted to have private fields shown. Not sure what we will do.

Reply:
Of course. It is private. Maybe say "The inner pointer" as you can't see the `ptr` field in the docs?

Reply:
Yeah, that is another option, definitely.

Reply:
> Of course. It is private. Maybe say "The inner pointer" as you can't see the `ptr` field in the docs?

That is sort of I what I did for rbtrees: https://github.com/Rust-for-Linux/linux/pull/400/files#diff-0cb585b4b0d3d30535d603fd0ae07753fcd5a180b13d8f9f06164093c4551722R40